src/main/java/de/tum/in/www1/artemis/service/LectureUnitProcessingService.java

```diff
@@ -113,10 +119,10 @@ private void removeBreakOrSolutionSlides(PDDocument document, boolean removeBrea
                 PDDocument currentPage = pages.get(index);
                 String slideText = pdfTextStripper.getText(currentPage);
 
-                if (isBreakSlide(slideText) && removeBreakSlides) {
+                if (slideContainsKeyword(slideText, removeBreakSlidesKeyword) && removeBreakSlides) {
```

Comment:
```suggestion
                if (removeBreakSlides && slideContainsKeyword(slideText, removeBreakSlidesKeyword)) {
```
performance improvement

Reply:
Done 506d2d9999f7ea6b72191d2e8d9774caf4db6955